control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3743,17 +3744,66 @@ func (r *HostedControlPlaneReconciler) reconcileOperatorLifecycleManager(ctx con
 				return fmt.Errorf("failed to reconcile catalogs: %w", err)
 			}
 
-			catalogsImageStream := manifests.CatalogsImageStream(hcp.Namespace)
-			if !overrideImages {
-				isImageRegistryOverrides := util.ConvertImageRegistryOverrideStringToMap(p.OLMCatalogsISRegistryOverridesAnnotation)
-				if _, err := createOrUpdate(ctx, r, catalogsImageStream, func() error {
-					return olm.ReconcileCatalogsImageStream(catalogsImageStream, p.OwnerRef, isImageRegistryOverrides)
-				}); err != nil {
-					return fmt.Errorf("failed to reconcile catalogs image stream: %w", err)
+			isImageRegistryOverrides := util.ConvertImageRegistryOverrideStringToMap(p.OLMCatalogsISRegistryOverridesAnnotation)
+
+			if r.ManagementClusterCapabilities.Has(capabilities.CapabilityImageStream) {
+				catalogsImageStream := manifests.CatalogsImageStream(hcp.Namespace)
+				if !overrideImages {
+					if _, err := createOrUpdate(ctx, r, catalogsImageStream, func() error {
+						return olm.ReconcileCatalogsImageStream(catalogsImageStream, p.OwnerRef, isImageRegistryOverrides)
+					}); err != nil {
+						return fmt.Errorf("failed to reconcile catalogs image stream: %w", err)
+					}
+				} else {
+					if _, err := util.DeleteIfNeeded(ctx, r, catalogsImageStream); err != nil {
+						return fmt.Errorf("failed to remove OLM Catalog ImageStream: %w", err)
+					}
 				}
-			} else if r.ManagementClusterCapabilities.Has(capabilities.CapabilityImageStream) {
-				if _, err := util.DeleteIfNeeded(ctx, r, catalogsImageStream); err != nil {
-					return fmt.Errorf("failed to remove OLM Catalog ImageStream: %w", err)
+			} else {
+				pullSecret := common.PullSecret(hcp.Namespace)
```

Comment:
should you not check here if overrideImages is set?

Reply:
No because this should also be run when there is no registry overrides

Reply:
But if overrideImages is set shouldn't we skip the logic of pulling from the registry?

Reply:
no we want to pull from the registry regardless of overrideImages. AKS has no ImageStream functionality which is what we use to set the catalog images so we need logic to do this on AKS regardless of the overrides. But if the overrides are provided then we should retrieve them catalog images from the mirror registry.

Reply:
Ah I gotcha now, was mixing it up with registryOverrides.